subsys/bluetooth/host/smp.h

```diff
@@ -178,3 +179,17 @@ int bt_smp_sign(struct bt_conn *conn, struct net_buf *buf);
 
 /** Generate IRK from Identity Root (IR) */
 int bt_smp_irk_get(uint8_t *ir, uint8_t *irk);
+
+/** Converts a SMP error to string.
+ *
+ * The error codes are described in the Bluetooth Core specification,
+ * Vol 3, Part H, Section 3.5.5.
+ *
+ * The Security Manager Protocol documentation found in Vol 4, Part H,
+ * describes when the different error codes are used.
+ *
+ * See also the defined BT_SMP_ERR_* macros.
+ *
+ * @return The string representation of the SMP error code.
+ */
+const char *bt_smp_err_to_str(uint8_t smp_err);
```

Comment:
Since this is now an internal function, are you planning on using it in the host somewhere?

Reply:
This seems like a good place to use it: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/smp.c#L1653

Reply:
Indeed :)